[mamayuan] ip

mamayuan opened the pull request:
## DukePro - Task Manager :book:
> “Your mind is for having ideas, not holding them.” – David Allen 

**DaDaYuan** _frees_ your mind of having to remember things you need to do. It's:
- **text-based**
- ~~complex~~ **easy** to learn
- **FAST SUPER FAST** to use :rocket:

All you need to do is:
1. [Download it from here](#)
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you :wink:

And it is ~~expensive~~ **FREE**! :tada:

### Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders

If you're a `Java` programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

mamayuan committed:
Rename, Greet, Exit

mamayuan committed:
Rename of file

mamayuan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

mamayuan committed:
Correct missing colon

mamayuan committed:
Echo

mamayuan committed:
Add, List

mamayuan committed:
Mark as Done

mamayuan committed:
ToDos, Events, Deadlines

mamayuan committed:
Handle Errors

mamayuan committed:
Delete

mamayuan committed:
Packages inner classes

mamayuan committed:
Reorganise packages

mamayuan committed:
Reorganises packages again

mamayuan committed:
Fix bugs up to level-5

mamayuan committed:
Merges branches

mamayuan committed:
fixes bugs up to level-6

mamayuan committed:
Adds Store

mamayuan committed:
Merge branch 'master' of https://github.com/mamayuan/ip into branch-Level-7

# Conflicts:
#	src/main/java/duke/io/Loader.java
#	src/main/java/duke/io/Saver.java

mamayuan committed:
Settles conflicts

mamayuan committed:
Settles conflicts

mamayuan committed:
Finalises level-7

mamayuan committed:
Adds TimeParser.java

mamayuan committed:
Adds OOP and Level-9

jibtaf commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+package duke;
+
+import duke.exceptions.FileIoException;
+import duke.tasks.*;
```
Should the classes imported be explicitly listed out?

jibtaf commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+import duke.exceptions.UnknownCommandException;
+
+
+import java.util.ArrayList;
```
I think `import java.*` should be before `import duke.*`, according to the checkStyle function.

jibtaf commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.UnknownCommandException;
+
+
```
Perhaps one blank line is enough?

jibtaf reviewed (commented):
Looks good to me, although there may be some style errors. Perhaps adding javadoc and header comments for public methods will make your code more clear and easier to understand. Also, according to the Git conventions, you should use "Add", "Settle", "Finalise" instead of "Adds", "Settles", and "Finalises". Overall, well done! 👍

mamayuan committed:
Fixes bugs

ChangruHenryQian commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+                        throw new Exception("OOPS!!! The description of a todo cannot be empty.");
+                    } else {
+                        tasks.addTask(new Todo(description));
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println("  " + tasks.getTask(tasks.getNumberOfTasks() - 1));
+                        System.out.println("Now you have " + tasks.getNumberOfTasks() + " task(s) in the list.");
```
Should Ui class do the printing part?

ChangruHenryQian commented on src/main/java/duke/tasks/TaskType.java:
```diff
@@ -0,0 +1,8 @@
+/**
+ * An enum class that specify the type of the task.
+ */
+public enum TaskType {
+    TODO, EVENT, DEADLINE
+}
```
I like your implementation of the enum class for task types.

ChangruHenryQian commented on src/main/java/duke/exceptions/ErrorMessages.java:
```diff
@@ -0,0 +1,31 @@
+package duke.exceptions;
+
+public class ErrorMessages {
```
Great implementation on ErrorMessage class.

ChangruHenryQian commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,124 @@
+package duke.io;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.tasks.Event;
+import duke.tasks.Deadline;
+import duke.tasks.TaskList;
+import duke.exceptions.UnknownCommandException;
+import duke.exceptions.FileIoException;
+import duke.exceptions.ErrorMessages;
```
I enjoy your clean separation and ordering of the import statements.

ChangruHenryQian reviewed (commented):
Good job!

mamayuan committed:
Add checkstyle

mamayuan committed:
Fix bugs on datetime

mamayuan committed:
[feature] Add UI Components and Launcher for Duke Application

The Duke application lacks a user interface and a main entry point for users to interact with.

To make the Duke application user-friendly and navigable, we need a proper user interface and an entry point to launch the application.

Implement the MainWindow class to act as the controller for the main application window.
Add the DialogBox custom control to display dialogues for the user and Duke with associated images.
Introduce the Launcher class as the application's entry point.

Using JavaFX, we can ensure a consistent and efficient user interface. The DialogBox custom control allows us to encapsulate dialogue functionalities, ensuring code reusability and organization. The Launcher acts as a clear entry point for the application, keeping the initialization process straightforward.

Added Javadoc comments to the newly created classes for better documentation and clarity.

mamayuan committed:
Add A-Varargs

mamayuan committed:
Add java docs

mamayuan committed:
Add java docs

mamayuan committed:
Add java doc

mamayuan committed:
Add Javadoc

mamayuan committed:
Add HelpCommand

mamayuan committed:
Add images for GUI

mamayuan committed:
Fix code quality

mamayuan committed:
Add Java docs

mamayuan committed:
Finalise GUI

mamayuan committed:
Remove unnecessary attributes

mamayuan committed:
Add A-Assertions

mamayuan committed:
Add Junit test files

mamayuan committed:
Fix Checkstyle

mamayuan committed:
Add User Guide

mamayuan committed:
Add image for Readme

mamayuan committed:
Fix Javfx version

mamayuan committed:
Gradle

mamayuan committed:
Fix bugs for Event storage

mamayuan committed:
Fix TimeParser bugs

mamayuan committed:
Test

mamayuan committed:
Merge pull request #2 from mamayuan/Test

Test

mamayuan committed:
Fix bugs

mamayuan committed:
Merge branch 'master' of https://github.com/mamayuan/ip

mamayuan committed:
Add Ui.png to readme

mamayuan committed:
Fix UserGuide

mamayuan committed:
Minor changes

mamayuan committed:
Add link to release
